here-naksha-lib-ext-manager/src/main/java/com/here/naksha/lib/extmanager/ExtensionCache.java

```diff
@@ -112,41 +112,66 @@ private void publishIntoCache(KVPair<Extension, File> result, ExtensionConfig ex
       if (!isNullOrEmpty(extension.getInitClassName())) {
         try {
           Class<?> clz = loader.loadClass(extension.getInitClassName());
-          clz.getConstructor(INaksha.class, Extension.class).newInstance(naksha, extension);
-        } catch (ClassNotFoundException
-            | InvocationTargetException
-            | InstantiationException
-            | NoSuchMethodException
-            | IllegalAccessException e) {
+          Object obj = clz.getConstructor().newInstance();
+          if (obj instanceof IExtensionInit initInstance) {
+            initInstance.init(naksha, extension);
+            instance = initInstance;
+          } else {
+            logger.error("Extension does not implement IExtensionInit for extension {}", extension.getId());
+            return;
+          }
+        } catch (Exception e) {
           logger.error(
               "Failed to instantiate class {} for extension {} ",
               extension.getInitClassName(),
-              extension.getId(),
+              extensionIdWthEnv,
               e);
           return;
         }
       }
       if (!isNullOrEmpty(extension.getInitClassName()))
         logger.info(
             "Extension {} initialization using initClassName {} done successfully.",
-            extension.getId(),
+            extensionIdWthEnv,
             extension.getInitClassName());
-      loaderCache.put(extension.getId(), new KVPair<Extension, ClassLoader>(extension, loader));
-      PluginCache.removeExtensionCache(extension.getId());
+
+      loaderCache.put(extensionIdWthEnv, new ValueTuple(extension, loader, instance));
+      PluginCache.removeExtensionCache(extensionIdWthEnv);
+
       logger.info(
           "Extension id={}, version={} is successfully loaded into the cache, using Jar at {} for env={}.",
```

Comment:
As extensionIdWthEnv already contains env now, we can remove explicitly logging "env"

Reply:
Done.